docs/DeveloperGuide.md

```diff
@@ -1,34 +1,614 @@
+list.
+
+Below shows the sequence diagram for ```AddAppointmentCommand``` class
+
+
+#### 2.2.5 Parser module
+This section describes the implementation of Parser class, as well as the design considerations and rational behind the 
+current implementation.
+
+The main purpose of the Parser class is as below.
+1.  To interpret user inputs so that the correct command can be executed 
+2.  Functions as the first line to sanitize user input
+
+As such, the Parser class only has one publicly callable method and that returns the command object for execution. 
+To assist the public method, Parser class has multiple private helper methods, as well as access to the Exception handler
+to ensure that the user input is formatted correctly. This way, purpose 1 and 2 is satisfied. 
+
+##### 2.2.5.1 Object creation and steps in input interpretation
+1.  The `Parser` object is first created in `Duke` class and subsequently used until program termination.
+2.  User input is received and handed over to `Parser` object for interpretation.
+3. In the `Parser` object, the type of command is first determined via helper method `getCommand(userInput)`.
+    + Example: `addp \age 23 \name Justin`, the `addp` command will be determined.
+4. The remaining fields will be recorded in a hashMap through either `fillPatientFields(userInput)` or `fillAppointmentFields(userInput)`
+depending on the command type in Step 3.
+    + The type of category of the command can be determined based on the last alphabet of the first word. 
+    + The command `addp` has the last alphabet is `p`, `fillPatientFields()` will be called. 
+    Likewise, if it ise `adda`, it will be `fillAppointmentFields()`.
+    + Example: `addp \name Sam \age 18`, the method `fillPatientFields(userInput)` will be called. The hashMap will contain
+    age -> 23, name -> Sam. 
+5. At the end of the execution, a reference to the command object will be returned. 
+
+>![](images/SD_parser/Capture2.JPG)
+
+>![](images/SD_parser/Capture.JPG)
+
+Sequence Diagram when `parseCommand(userInput)` is initially called
+![](images/SD_parser/Slide1.JPG)
+
+Sequence Diagram for `addp`
+![](images/SD_parser/Slide2.JPG)
+
+Sequence Diagram for `editp`
+![](images/SD_parser/Slide3.JPG)
+
+Sequence Diagram for `deletep`
+![](images/SD_parser/Slide4.JPG)
+
+Sequence Diagram for `adda`
+![](images/SD_parser/Slide5.JPG)
+
+Sequence Diagram for `edita`
+![](images/SD_parser/Slide6.JPG)
+
+Sequence Diagram for `deletea`
+![](images/SD_parser/Slide7.JPG)
+
+Sequence Diagram when it is an unknown command
+![](images/SD_parser/Slide8.JPG)
+
+Sequence Diagram for the creation of the command Object
+
+![](images/SD_parser/Slide9.JPG)
+
+Sequence Diagram for error checking when `DukeExpcetion` is called
+
+![](images/SD_parser/Slide10.JPG)
+
+Sequence Diagram for calling an enum
+
+![](images/SD_parser/Slide11.JPG)
+
+Sequence Diagram for error checking when `DukeExpcetion` is called
```

Comment:
![image](https://user-images.githubusercontent.com/49268567/77993330-842cc980-735a-11ea-87e1-eb215db0e50a.png)
The `<<sth>>` is used in UML to denote enumeration. Maybe you can reconsider the symbol here?